qiskit/pulse/instructions/instruction.py

```diff
@@ -275,7 +276,8 @@ def assign_parameters(self,
             value = value_dict[parameter]
             op_indices = self._parameter_table[parameter]
             for op_idx in op_indices:
-                new_operands[op_idx] = new_operands[op_idx].assign(parameter, value)
+                param_expr = new_operands[op_idx]
+                new_operands[op_idx] = format_parameter_value(param_expr.assign(parameter, value))
```

Comment:
Though all test have passed, this could cause problem in parameterization of channel index because it seems like evaluating timeslot update based on value type (in Schedule timeslot is updated if channel index value is ParameterExpression but this format value before the evaluation).

Reply:
I don't quite follow. Could you write a test that would cause this to break?

Reply:
I carefully read the code and concluded this is not a problem. Because channel index is parameterized at schedule level, so converting ParameterExpression into formatted type at instruction level doesn't affect channel parameterization (channel index is still ParameterExpression).